validator/rules/example_matches.go

```diff
@@ -0,0 +1,45 @@
+package rules
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/aerogear/charmil/validator"
+	"github.com/spf13/cobra"
+)
+
+// define errors for the UseMatches Rule
+var (
+	ErrExampleMistmatch = errors.New("example mismatch")
+)
+
+var ExampleMatchesRule = "EXAMPLE_MATCHES_RULE"
+
+// ExampleMatches defines Regexp to be compared
+type ExampleMatches struct {
+	RuleOptions validator.RuleOptions
+}
+
+// Validate is a method of type Rule Interface
+// which returns validation errors
+// checks if the example is updated as per command
+func (e *ExampleMatches) Validate(cmd *cobra.Command) []validator.ValidationError {
+	var errors []validator.ValidationError
+
+	// if command needs to be ignored
+	if val, ok := e.RuleOptions.SkipCommands[cmd.CommandPath()]; ok {
+		if val {
+			return errors
+		}
+	}
+
+	cmdPath := cmd.CommandPath()
+	cmdExample := cmd.Example
+
+	// check if cmdPath is not the substring of cmdExample
+	if cmdPath != "" && cmdExample != "" && !strings.Contains(cmdExample, cmdPath) {
+		errors = append(errors, validator.ValidationError{Name: "provided example doesn't correlate with the command", Err: ErrExampleMistmatch, Rule: ExampleMatchesRule, Cmd: cmd})
```

Comment:
Sorry for being annoying but we need to be a bit more concrete here, so they know what to do to fix this error. 

Maybe something like "provided example doesn't contain command path -> %cmdExample : %cmdPath"